### PR TITLE
CFE-2524: Variable table copying amousset cfe2524

### PR DESCRIPTION
**Pull request by jimis**
Continues from where #2818 was.

### CODE REVIEW

**Comment by jimis** [comment]
If all tests go fine, I'll squash and merge with your commit @amousset if that's OK with you.

**Comment by jimis** [comment]
Simplified your testcase in the last commit, @amousset .

**Comment by jimis** [comment]
Wow all ENT tests are green! Please have a look before I squash as much as possible.

**Comment by ncharles** [comment]
<3 thank you @jimis @amousset

**Comment by amousset** [comment]
ncf tests are green too!

File: libpromises/expand.c [hunk 1]
```diff
@@ -256,7 +256,8 @@ PromiseResult ExpandPromise(EvalContext *ctx, const Promise *pp,
      *    (including body inheritance). */
     Promise *pcopy = DeRefCopyPromise(ctx, pp);
 
-    EvalContextStackPushPromiseFrame(ctx, pcopy, true);
+    // TODO: Remove last parameter?
```
Comment (amousset) [inline libpromises/expand.c]: we can remove this TODO

Comment (amousset) [inline libpromises/expand.c]: we can remove this TODO

Comment (jimis) [reply libpromises/expand.c]: removing it, thanks

**Pull request merged**